### PR TITLE
Support to create staff users on first login via Oauth

### PR DESCRIPTION
**Pull request by diegobz**
I'm giving this a shot and maybe it's something we can incorporate upstream after maturing the idea a bit.

Opening this PR mostly to gather feedback for now.

### CODE REVIEW

**Comment by matthiask** [comment]
Thank you! It's an interesting idea. The proposed implementation would not work for us right now because we often use custom user models which do not have username fields, only email, and autocreation would not work as proposed here.

**Comment by diegobz** [comment]
@matthiask I see. Maybe we could also have an option mapping fields that should be used in the Creation of `User` objects. I will need to think a bit about it.

**Comment by diegobz** [comment]
How about using `<UserModel>.objects.create_user(**kwargs)`, so even if `username` is passed, you can ignore it on your code?

File: admin_sso/auth.py [hunk 1]
```diff
@@ -1,9 +1,49 @@
 from django.contrib.auth import get_user_model
-
 from admin_sso.models import Assignment
 
 
 class DjangoSSOAuthBackend(object):
+
+    def _get_username(self, sso_email):
```
Comment (diegobz) [inline admin_sso/auth.py]: @matthiask How about this solution? You can overwrite this particular method and adjust things in the way you expect it in the field that's set as unique in your custom user model.

Based on https://docs.djangoproject.com/en/1.10/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD

**Comment by matthiask** [comment]
Sorry for not getting back to you sooner!

I suspect you'd have an easier time building on https://github.com/matthiask/django-authlib instead of django-admin-sso if you need this functionality. I built a less capable replacement for django-admin-sso which might serve as an example how easy OAuth2 authentication for the admin panel is built from the ground up:

https://github.com/matthiask/django-authlib/blob/master/authlib/admin_oauth/views.py

I'm not convinced that automatic account creation is a good fit for django-admin-sso, but it should be relatively simple to build the same functionality on top of authlib... what do you think?

**Comment by diegobz** [comment]
@matthiask I will check once I get some spare time.

**Pull request closed without merging**